framewirc/messages.py

```diff
@@ -79,37 +79,35 @@ def _chunk_message(message, max_length):
     lines = deque(message.splitlines())
     while lines:
         line = lines.popleft()
-        line_bytes = to_bytes(line)
+        line_bytes = line.encode()
         # If the line fits, add it the the lines.
         if len(line_bytes) < max_length:
             yield line_bytes
             continue
 
-        # Whole line doesn't fit, so see if it can be split on space.
-        letterpoint = None  # Where we should break if there is no space
-        spacepoint = None  # Where we should break if there is a space.
+        # See if there is a space below the max length.
+        spacepoint = line_bytes.rfind(b' ', 0, max_length+1)
+        if spacepoint != -1:
+            # Break on the last space that fits.
+            start = line_bytes[:spacepoint]
+            yield start
+            # ... and add what's left back into the line pool.
+            end = line_bytes[(spacepoint + 1):].decode()
+            lines.appendleft(end)
+            # And move onto the next line.
+            continue
+
+        # Whole line does not contain spaces, so split within word.
         line_length = 0  # The running total size of the line
         for i, str_char in enumerate(line):
-            char_length = len(to_bytes(str_char))
+            char_length = len(str_char.encode())
             line_length += char_length
-            if str_char == ' ':
-                spacepoint = i
             if line_length > max_length:
                 letterpoint = i
                 break
```

Comment:
What a great idea! You're right, I'm yielding utf8-encoded bytes, and this will be a much better solution than my binary search proposal. I'll have a play with this and see what I come up with. Thank you!